qemu/tests/boot_order_check.py

```diff
@@ -73,17 +73,16 @@ def run(test, params, env):
     boot_fail_infos = boot_fail_infos % (list_nic_addr[0][0],
                                          list_nic_addr[1][0],
                                          list_nic_addr[2][0])
-
-    error.context("Check the guest boot result", logging.info)
     if bootorder_type == "type2":
-        session_serial = vm.wait_for_serial_login(timeout=timeout)
-        output = vm.serial_console.get_stripped_output()
-        session_serial.close()
+        boot_grap_str = "Booting from Hard Disk.*"
     else:
-        output = vm.serial_console.get_stripped_output()
-        utils_misc.wait_for(lambda: re.search("No bootable device.", output),
-                            timeout, 1)
-        output = vm.serial_console.get_stripped_output()
+        boot_grap_str = "No bootable device."
+
+    error.context("Check the guest boot result", logging.info)
+    output = vm.serial_console.get_stripped_output()
+    utils_misc.wait_for(lambda: re.search(boot_grap_str, output),
+                        timeout, 1)
+    output = vm.serial_console.get_stripped_output()
```

Comment:
It is meaningless from L76 to L84, we can just make it as:
```
error.context("Check the guest boot result", logging.info)
output = vm.serial_console.get_stripped_output()
```

Reply:
Thanks. `vm.serial_console.get_stripped_output()` returns once it gets string from serial console, and serial console can get continuous output when seabios starts, so Line83 it to check if there is enough log to do `re.findall()` in Line94.

Reply:
Not exactly, it works well only if the message which it attempts to check is appeared in the `output` which is got at the first time (L82), otherwise it is totally wasting time at L83 because `output` will never be updated since then.

The right solution is to directly check `boot_fail_infos` in a loop and get the `output` every time.

Reply:
Thanks, updated it, could you help review it again?

 (1/3) smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_order_check.bootorder0: PASS (227.37 s)
 (2/3) smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_order_check.bootorder1: PASS (227.35 s)
 (3/3) smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_order_check.bootorder2: PASS (228.25 s)